### ISSUE TITLE
Pulumi CLI panics on `preview` when trying to create a new `aws.Provider` for a new `aws.organizations.Account`

### ISSUE THREAD

**Issue by tom-fletcher** [pulumi/pulumi-aws#2818]
### What happened?

When trying to create a new `aws.organizations.Account`, and assign a new `aws.Provider` to that account with an assumed role, the CLI exits with `panic: runtime error: invalid memory address or nil pointer dereference`.

This prevents creating new organizations accounts and immediately deploying resources into that account.

If the account has been created previously, a provider will be created as expected.

### Example

A minimal repro can be found here: https://github.com/tom-fletcher/pulumi-awsprovider-repro

The infrastructure code in this repro is:

```ts
import * as pulumi from '@pulumi/pulumi'
import * as aws from '@pulumi/aws'

const account = new aws.organizations.Account(
    'account',
    {
        email: 'test@test.com',
        closeOnDeletion: true,
    }
);

const provider = new aws.Provider(
    'provider',
    {
        region: 'us-east-1',
        assumeRole: {
            roleArn: pulumi.interpolate`arn:aws:iam::${account.id}:role/OrganizationAccountAccessRole`,
        },
    }
);

// This will not be created because `pulumi preview` panics when trying to create the provider
new aws.s3.Bucket('bucket', {}, { provider })
```

The CLI panic output on `pulumi up` is:

```
  pulumi:pulumi:Stack (awsprovider-repro-repro):
    panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x130e43b]
    goroutine 53 [running]:
    github.com/hashicorp/terraform-provider-aws/internal/framework/types.durationType.Validate(0x0?, {0x1a1a3240?, 0x0?}, {{0x0, 0x0}, {0x0, 0x0}}, {{0xc00ae873c0, 0x3, 0x4}})
        /home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/framework/types/duration.go:106 +0x7b
    github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.Data.ValueAtPath({{0xe1937c2, 0xd}, {0xf6f9eb8, 0xc001b63e80}, {{0xf6ec4b8, 0xc0034dfbf0}, {0xc671640, 0xc0034deb70}}}, {0xf6dc250, 0xc0036342a0}, ...)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwschemadata/data_value.go:82 +0x675
    github.com/hashicorp/terraform-plugin-framework/internal/fwserver.AttributeValidate({0xf6dc250, 0xc003467c80}, {0xf6f2d00, 0xc001b799e0}, {{{0xc00ae873c0, 0x3, 0x4}}, {0x1, {0xc00ae87400, 0x3, ...}}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/attribute_validation.go:74 +0x275
    github.com/hashicorp/terraform-plugin-framework/internal/fwserver.NestedBlockObjectValidate({0xf6dc250, 0xc003467c80}, {0xf6e4d40, 0xc00ae86c80}, {{{0xc00adb1c20, 0x2, 0x2}}, {0x1, {0xc00adb1c40, 0x2, ...}}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/block_validation.go:439 +0x985
    github.com/hashicorp/terraform-plugin-framework/internal/fwserver.BlockValidate({0xf6dc250, 0xc003467c80}, {0xf6eccc8, 0xc001bc07e0}, {{{0xc00ae81ac0, 0x1, 0x1}}, {0x1, {0xc00ae81ae0, 0x1, ...}}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/block_validation.go:83 +0xf45
    github.com/hashicorp/terraform-plugin-framework/internal/fwserver.SchemaValidate({0xf6dc250, 0xc003467c80}, {0xf6f9eb8, 0xc001b63e80}, {{{{0xf6ec4b8, 0xc0034dfbf0}, {0xc671640, 0xc0034deb70}}, {0xf6f9eb8, 0xc001b63e80}}}, ...)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/schema_validation.go:66 +0x485
    github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ValidateProviderConfig(0xc0008fcf20, {0xf6dc250, 0xc003467c80}, 0xc0020085c0, 0xc001bcfa10)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/server_validateproviderconfig.go:87 +0x390
    github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ValidateProviderConfig(0xc0008fcf20, {0xf6dc250?, 0xc003404750?}, 0x0?)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/proto6server/server_validateproviderconfig.go:39 +0x1dc
    github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge.(*provider).validateProviderConfig(0xc0009e4100, {0xf6dc250, 0xc003404750}, {0xc00aaa4c80, 0x43}, 0x0?)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/pf@v0.16.1/tfbridge/provider_checkconfig.go:111 +0x197
    github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge.(*provider).CheckConfigWithContext(0xc0009e4100, {0xf6dc250?, 0xc0034042a0?}, {0xc00aaa4c80, 0x43}, 0xc9fc2e0?, 0xc003404540, 0x0?)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/pf@v0.16.1/tfbridge/provider_checkconfig.go:86 +0x413
    github.com/pulumi/pulumi-terraform-bridge/pf/internal/plugin.(*providerServer).CheckConfig(0xc0033ca600, {0xf6dc250, 0xc0034042a0}, 0xc00abe53e0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/pf@v0.16.1/internal/plugin/provider_server.go:198 +0x1ad
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).CheckConfig.func1()
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:115 +0x6f
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.asyncJoin[...].func1()
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:562 +0x42
    created by github.com/pulumi/pulumi-terraform-bridge/x/muxer.asyncJoin[...] in goroutine 51
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:561 +0x6c

  pulumi:providers:aws (provider):
    error: rpc error: code = Unavailable desc = error reading from server: EOF
```

### Output of `pulumi about`

```
CLI          
Version      3.83.0
Go Version   go1.21.1
Go Compiler  gc

Plugins
NAME    VERSION
aws     6.2.0
nodejs  unknown

Host     
OS       ubuntu
Version  22.04
Arch     x86_64
```

### Additional context

A downgrade to `"@pulumi/aws": "5.42.0"` deploys the resources as expected. This issue appears limited to `6.x.x`.

Possibly, `pulumi preview` is trying to validate the `arn` exists, which it cannot as the account has not yet been created?

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by tom-fletcher** [comment]
A note as well to assist with triage - what makes this a bit nasty is that since the stack deploys normally if the `aws.organizations.Account` resource already exists in the stack, the bug remains hidden up and until you try to create a new account.

So if you have upgraded your stack from `v5.x` to `v6.x`, and you are using Pulumi to auto-create accounts (say developer accounts, or customer accounts) then everything looks like it is working exactly as it should be, and you get on with integrating the new `v6.x` resources. At some point, maybe  a few months later, you go to onboard a new developer/customer/etc - and it is at this point the CLI blows up on you.

**Comment by mikhailshilkov** [comment]
Thank you for the issue and a full repro @tom-fletcher! So much appreciated.

I can reproduce the error locally. Looks like a TFPF integration error, @t0yv0 could you please take a look?

**Comment by eneiss** [comment]
Same issue is happening to me here, upgrading from 5.42.0 to 6.2.0 and I got panic errors too.
It seems that `pulumi preview` is using a lot of memory, reaching my 7.5 GB memory limit (3 GB used when idle).

After 2 reboots I managed to get the `pulumi preview` step to succeed but when following the memory usage with htop, it seems I barely avoided an OOM panic because my memory almost reached its maximum usage right before the process ended.
My previous panic errors were likely caused by OOM too, I followed the process execution with htop too and the error happened when the memory was full.

**Comment by t0yv0** [comment]
@eneiss could you open a separate issue with the panics you're getting and increased memory usage? I'm assuming that the example program listed in this ticket does not trigger the OOM or does it? Thank you.

**Comment by tom-fletcher** [comment]
@t0yv0 @eneiss I am not seeing any memory issues in the repro docker container with `htop` or `top`, so I suspect the OOM is a seperate issue.

**Comment by t0yv0** [comment]
Appreciate patience here while we decide on a fix here.

It appears to be a bug in plugin terraform framework fixable with this https://github.com/hashicorp/terraform-plugin-framework/compare/main...t0yv0:terraform-plugin-framework:t0yv0/fix-value-at-path-panic ; quite possibly this is not activated when using the TF provider under Terraform CLI due to a difference in how the unknown values are encoded and passed in. 

Our options so far are:

- build against the above patch

- disable provider unknown validation for PF, possibly as an option for AWS provider only; since our current architecture mixes SDKv2 and PF based providers and broadcasts configuration calls, that's not as bad as otherwise since the SDK2 provider will still perform validations on the values 
 
- possibly investigate deeper to the discrepancy between how the provider is invoked under TF and Pulumi

We'll aim to have something available by EOW to unblock here.

**Comment by t0yv0** [comment]
@tom-fletcher we believe v6.2.1 should fix the issue but please reopen if you are still having problems with it! Thank you.
